src/search/pdbs/pattern_collection_generator_hillclimbing.cc

```diff
@@ -588,86 +587,87 @@ static basic_string<char> paper_references() {
-        "optimized for the Evaluator#Canonical_PDB heuristic. It it described "
-        "in the following paper:" + paper_references());
-    add_hillclimbing_options(parser);
-
+    {
+        parser.document_synopsis(
+            "Hill climbing",
+            "This algorithm uses hill climbing to generate patterns "
+            "optimized for the Evaluator#Canonical_PDB heuristic. It it described "
+            "in the following paper:" + paper_references());
+        add_hillclimbing_options(parser);
+    }
     Options opts = parser.parse();
     if (parser.help_mode())
         return nullptr;
-
     check_hillclimbing_options(parser, opts);
     if (parser.dry_run())
         return nullptr;
-
     return make_shared<PatternCollectionGeneratorHillclimbing>(opts);
 }
 
-static shared_ptr<Heuristic> _parse_ipdb(OptionParser &parser) {
-    parser.document_synopsis(
-        "iPDB",
-        "This approach is a combination of using the Evaluator#Canonical_PDB "
-        "heuristic over patterns computed with the "
-        "PatternCollectionGenerator#hillclimbing algorithm for pattern "
-        "generation. It is a short-hand for the command-line option "
-        "{{{cpdbs(hillclimbing())}}}. "
-        "Both the heuristic and the pattern generation algorithm are described "
-        "in the following paper:" + paper_references() +
-        "See also Evaluator#Canonical_PDB and "
-        "PatternCollectionGenerator#Hill_climbing for more details.");
-
-    parser.document_language_support("action costs", "supported");
-    parser.document_language_support("conditional effects", "not supported");
-    parser.document_language_support("axioms", "not supported");
-    parser.document_property("admissible", "yes");
-    parser.document_property("consistent", "yes");
-    parser.document_property("safe", "yes");
-    parser.document_property("preferred operators", "no");
-
-    add_hillclimbing_options(parser);
+static Plugin<PatternCollectionGenerator> _plugin("hillclimbing", _parse);
 
-    /*
-      Add, possibly among others, the options for dominance pruning.
+static shared_ptr<Heuristic> _parse_ipdb(OptionParser &parser) {
+    {
+        parser.document_synopsis(
+            "iPDB",
+            "This approach is a combination of using the Evaluator#Canonical_PDB "
+            "heuristic over patterns computed with the "
+            "PatternCollectionGenerator#hillclimbing algorithm for pattern "
+            "generation. It is a short-hand for the command-line option "
+            "{{{cpdbs(hillclimbing())}}}. "
+            "Both the heuristic and the pattern generation algorithm are described "
+            "in the following paper:" + paper_references() +
+            "See also Evaluator#Canonical_PDB and "
+            "PatternCollectionGenerator#Hill_climbing for more details.");
+
+        add_hillclimbing_options(parser);
+        /*
+          Add, possibly among others, the options for dominance pruning.
```

Comment:
I would remove the empty line below because it interrupts the grouping (all options in one block).

Reply:
I don't mind much either way, I read a block comment as "one thing", so empty lines inside of it don't bother me, but I also don't think the empty line is necessary, so I removed it.